roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,49 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerized
+
+class DockerImageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+
+        args = {
+            "name": self.rpm_docker_images(),
+        }
+
+        docker_images = self.module_executor("docker_image_facts", args, tmp, task_vars)
+
+        if len(docker_images['images']) == 0:
+            return {"failed": True, "msg": "Unable to find any docker images!"}
+
+        # if we find any docker images, attempt to do 1-to-1 match between
+        # each image description name key and items from rpm_docker_images
+        matched_images = set()
+        for i in docker_images['images']:
+            repoTag = i['RepoTags'][0]
+
+            # check image version
+            tag = repoTag.split(':')
+            if tag[1] == openshift_release and repoTag[0] in self.rpm_docker_images():
+                matched_images.add(tag[0])
+
+        missing_images = set(self.rpm_docker_images()) - matched_images
+        if len(missing_images) > 0:
+            return {"failed": True, "msg": "There are missing docker images or images that did not match the current OpenShift version (%s): %s:\n" % (openshift_release, missing_images)}
+
+        return {"Changed": False}
```

Comment:
In modules, I think "changed" should be all lowercase. Here it doesn't really matter, we don't even need it.

For now we only look for a "failed" key: https://github.com/openshift/openshift-ansible/blob/f0a32af0548eb309b9bb3bb2c366d35bdfab1847/roles/openshift_health_checker/action_plugins/openshift_health_check.py#L73-L75